crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -867,6 +920,33 @@ mod tests {
             VsockPacket::from_rx_virtq_chain(&mem, &mut chain, MAX_PKT_BUF_SIZE).unwrap_err(),
             Error::DescriptorChainTooShort
         );
+
+        // Let's also test a valid descriptor chain, with both header and data on a single
+        // descriptor.
+        let v = vec![Descriptor::new(
+            0x5_0000,
+            PKT_HEADER_SIZE as u32 + 0x100,
+            VRING_DESC_F_WRITE as u16,
+            0,
+        )];
+        let mut chain = queue.build_desc_chain(&v[..1]).unwrap();
```

Comment:
nit: you can pass `&v` directly.

Reply:
oh right, copy&past from other tests, I think we can pass `&v` everywhere.
I can send a cleanup on top of this.

Reply:
Yes, right. I noticed that in the other tests as well, but didn't want to burden you to change it all 😆. On the other hand I also don't want to propagate the mistake to other places.

Reply:
No problem at all.
I'll fix the tests added here and fix the other in a separate commit.
Should I integrate that commit in this PR, or better a new one?

Reply:
Works either way as long as it's a separate commit. Whichever makes it easier for you.

Reply:
Okay, I'll include here, less noise for others ;-)